### PR TITLE
Fix API List DAGs inconsistent with UI/CLI

### PR DESCRIPTION
**Pull request by jpyen**
related: #16228

`/api/v1/dags` should yield the same output as the Web UI and CLI, namely:
- show all dags
- that the user has access to
- which are present as DAG (can be scheduled, source file available)

The current implementation does only list DAGs that the user has access to, but fails to get the true DAG bag list. Which means the API endpoint is showing many more, deleted DAGs, that the UI and Web CLI do not.

### CODE REVIEW

**Review by ephraimbuddy — CHANGES_REQUESTED** [review]
Can you add a test to avoid regression?

**Comment by jpyen** [comment]
> Can you add a test to avoid regression?

Sure thing, although I am not too sure how to reproduce this state/ explicitly test for this state. Do you have any recommendation?

**Comment by ephraimbuddy** [comment]
> > Can you add a test to avoid regression?
> 
> Sure thing, although I am not too sure how to reproduce this state/ explicitly test for this state. Do you have any recommendation?

I think you can achieve it by marking dag inactive. DagModel.is_active exists, so many mark a dag inactive and another active

File: tests/api_connexion/endpoints/test_dag_endpoint.py
Comment (ephraimbuddy) [inline tests/api_connexion/endpoints/test_dag_endpoint.py:402]: How about you add a query that asserts that the number of dags in DagModel is  3

Comment (jpyen) [reply tests/api_connexion/endpoints/test_dag_endpoint.py:402]: Number of DAGs in DagModel should exclude inactive DAGs, as far as I understand. In my opinion are inactive DAGs `.trash` and play no significant role.

**Comment by jpyen** [comment]
@ephraimbuddy Is it possible to run the full CI suite?

File: tests/api_connexion/endpoints/test_dag_endpoint.py
Comment (ephraimbuddy) [reply tests/api_connexion/endpoints/test_dag_endpoint.py:402]: I know but how are we sure that an inactive dag exists here? Do we have a test for the new method you created? I think that querying DagModel and asserting that it has 3 dags(both active and inactive) at this point will help to know that the method you created works fine. If there's an alteration in the method in the future to work another way, it'll be noticed. What do you think?

**Comment by ephraimbuddy** [comment]
> @ephraimbuddy Is it possible to run the full CI suite?

Once approved, it'll run more tests if needed

File: tests/api_connexion/endpoints/test_dag_endpoint.py
Comment (jpyen) [reply tests/api_connexion/endpoints/test_dag_endpoint.py:400]: That makes sense, should I query DagModel at the same place where we are testing the API Endpoint?

**Comment by jpyen** [comment]
Added @ephraimbuddy suggestions & rebased to master.

**Review by ephraimbuddy — APPROVED** [review]

**Comment by jpyen** [comment]
@kaxil Changes approved, checks passed and ready to merge. Thanks for the patience.

**Comment by jpyen** [comment]
Thanks @ephraimbuddy!

**Comment by ephraimbuddy** [comment]
Congrats on the PR @jpyen

**Pull request merged**
